pandas/core/generic.py

```diff
@@ -223,6 +224,12 @@
     from pandas.core.indexers.objects import BaseIndexer
     from pandas.core.resample import Resampler
 
+    if sys.version_info >= (3, 10):
+        from typing import Concatenate
+
+    else:
+        from typing_extensions import Concatenate
```

Comment:
It might be better to move that to pandas._typing in case it will be used in other files in the future